toolkit/tools/bldtracker/bldtracker.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Tool to initialize a csv file or record a new timestamp
+// for shell scripts during the image-building process.
+
+package main
+
+import (
+	"os"
+	"time"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/exe"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/logger"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/timestamp"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	app          = kingpin.New("bldtracker", "A tool that helps track build time of different steps in the makefile.")
+	scriptName   = app.Flag("script-name", "The name of the current tool.").Required().String()
+	stepName     = app.Flag("step-name", "The name of the current step.").Required().String()
+	actionName   = app.Flag("action-name", "The name of the current action.").Default("").String()
+	dirPath      = app.Flag("dir-path", "The folder that stores timestamp csvs.").Required().ExistingDir() // currently must be absolute
+	logPath      = app.Flag("log-path", "Directory for log files").Required().ExistingDir()
+	validModes   = []string{"n", "r"}
+	mode         = app.Flag("mode", "The mode of this tool. Could be 'initialize' ('n') or 'record'('r').").Required().Enum(validModes...)
+	completePath string
+)
+
+func main() {
+	app.Version(exe.ToolkitVersion)
+	kingpin.MustParse(app.Parse(os.Args[1:]))
+	logger.InitBestEffort(*logPath+"/chroot_timestamp.log", "trace")
```

Comment:
This seems inconsistent with how we initialize the logger elsewhere. Why the additional `"/chroot_timestamp.log"`?

Reply:
This enables bldtracker.go to use logger.Log functions, mostly to record error messages. "logPath" is obtained from create_worker_chroot.sh. The log file for this shell script is also not in any of the folders under build/logs, so I mimicked that and added /chroot_timestamp.log. Would you recommend storing all of the logs generated by timestamping create_worker_chroot.sh in build/log/worker_chroot.log?

Reply:
Ah, I think I see what you mean. Here's how I'd approach this:

1. Tools shouldn't make assumptions about who and how they are called. If the caller passes a log path, then it would be confusing for the caller to see the logs appear elsewhere.
2. We already handle `create_worker_chroot.sh` logging by piping/redirecting everything we care to log into `$chroot_log`. Since we would be interested in any errors from `bldtracker`, I'd pipe its output in the same way, so it ends up in the same log file. This will also allow us to split the implementation of `bldtracker` from how it's called inside `create_worker_chroot.sh`.
3. We can still keep a separate log file just for `bldtracker`. Since this is a small helper tool and not one of the "beefier" ones, which constitute a separate build step all by themselves, I'd be also fine with not maintaining a separate log file for `bldtracker`.

I'm glad you asked this question btw. - I missed analyzing how the tool is used inside `create_worker_chroot.sh`.

Reply:
Thank you for the detailed answer! I think I will put the `bldtracker` logs in `$chroot_log` for `create_worker_chroot.sh` and not create a separate log file for `bldtracker`, since then I am logging the same issues twice.